src/Microsoft.Framework.Runtime.Abstractions/Infrastructure/HashCodeCombiner.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Collections;
+using System.Collections.Generic;
+
+namespace Microsoft.Framework.Runtime.Infrastructure
+{
+    internal class HashCodeCombiner
```

Comment:
See related issue aspnet/Razor#449. Discussion there was about whether any change was needed. But if the class is useful elsewhere, a shared project makes sense.